homeassistant/components/fritzbox/config_flow.py

```diff
@@ -110,12 +114,20 @@ async def async_step_ssdp(self, user_input):
         host = urlparse(user_input[ATTR_SSDP_LOCATION]).hostname
         self.context[CONF_HOST] = host
 
+        uuid = user_input.get(ATTR_UPNP_UDN)
+        if uuid:
+            if uuid.startswith("uuid:"):
+                uuid = uuid[5:]
+            await self.async_set_unique_id(uuid)
```

Comment:
You should also abort if the unique ID is already configured.

```suggestion
            await self.async_set_unique_id(uuid)
            self._abort_if_unique_id_configured({ CONF_HOST: host })
```